content/docs/instrumenting/writing_exporters.md

```diff
@@ -5,67 +5,46 @@ sort_rank: 5
 
 # Writing exporters
 
-When directly instrumenting your own code, the general rules of how to
-instrument code with a Prometheus client library can be followed quite
-directly. When taking metrics from another monitoring or instrumentation
-system, things tend not to be so black and white.
+If you are instrumenting your own code, the [general rules of how to
+instrument code with a Prometheus client library](/docs/practices/instrumentation/) should be followed. When taking metrics from another monitoring or instrumentation system, things tend not to be so black and white.
 
 This document contains things you should consider when writing an exporter or
 custom collector. The theory covered will also be of interest to those doing
 direct instrumentation.
 
-If you are writing an exporter and are unclear on anything here, contact us on
-IRC (#prometheus on Freenode) or the [mailing list](/community).
+If you are writing an exporter and are unclear on anything here, please contact us on IRC (#prometheus on Freenode) or the [mailing list](/community).
 
 ## Maintainability and purity
 
 The main decision you need to make when writing an exporter is how much work
 you’re willing to put in to get perfect metrics out of it.
 
 If the system in question has only a handful of metrics that rarely change,
-then getting everything perfect is an easy choice (e.g. the [haproxy
-exporter](https://github.com/prometheus/haproxy_exporter)).
+then getting everything perfect is an easy choice, a good example of this is the [HAProxy exporter](https://github.com/prometheus/haproxy_exporter).
 
-If on the other hand the system has hundreds of metrics that change
-continuously with new versions, if you try to get things perfect then you’ve
-signed yourself up for a lot of ongoing work. The [mysql
-exporter](https://github.com/prometheus/mysqld_exporter) is on this end of the
+If, on the other hand, the system has hundreds of metrics that change
```

Comment:
I'd rephrase this a little more to get rid of the double if, like:

> On the other hand, if you try to get things perfect when the system has hundreds of metrics that change frequently with new versions, then you’ve signed yourself up for a lot of ongoing work.

Reply:
Yes - thanks much nicer.